examples/1d_stencil/stencil.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 Nikunj Gupta
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/include/compute.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/async.hpp>
+#include <hpx/include/components.hpp>
+#include <hpx/include/parallel_algorithm.hpp>
+#include <hpx/include/util.hpp>
+
+
+#include <algorithm>
+#include <array>
+
+using allocator_type = hpx::compute::host::block_allocator<double>;
+using data_type = hpx::compute::vector<double, allocator_type>;
+
+const double k = 0.5;     // heat transfer coefficient
+const double dt = 1.;     // time step
+const double dx = 1.;     // grid spacing
+
+template <typename Container>
+void init(
+    std::array<Container, 2>& U, std::size_t Nx,
+    std::size_t rank = 0, std::size_t num_localities = 1)
+{
+    // Initialize: Boundaries are set to 1, interior is 0
+    if (rank == 0)
+    {
+        U[0][0] = 1.0;
+        U[1][0] = 1.0;
+    }
+    if (rank == num_localities - 1)
+    {
+        U[0][Nx-1] = 100.0;
+        U[1][Nx-1] = 100.0;
+    }
+}
+
+void stencil_update(std::array<data_type, 2>& U, const std::size_t& begin
+        , const std::size_t& end, const std::size_t t)
+{
+    data_type& curr = U[t % 2];
+    data_type& next = U[(t + 1) % 2];
+
+    #pragma unroll
```

Comment:
@NK-Nikunj this might be interesting for you: https://cdash.cscs.ch//viewBuildError.php?buildid=112799. The loop definitely looks unrollable to me but I might be missing something. The build is in release mode, so loop unrolling shouldn't be disabled because of a debug build at least. Do you know if GCC manages to unroll this (it might just not be reporting that it can't like clang)?

Reply:
Interesting. I didn't know that it was unrollable for certain builds. GCC manages to unroll just fine for me on my system.

Reply:
I think besides this we could go ahead with this PR. You can always improve it at a later point with another pull request. Or did you have other improvements in the pipeline still?

Would it be acceptable to remove the `#pragma unroll` does performance change significantly without it? Some builders do not seem to recognize the pragma, and some (like clang) don't want to unroll it. That leaves two options: figure out which compilers support the pragma and which need to have the warning silenced and then put a bunch of `ifdef` to make the builds pass, or remove the pragma. The former would be better, but the latter is definitely simpler if the pragma doesn't actually make a big difference.

Reply:
I have one optimization in mind. The asynchronous nature of set with low grain size adds contention. I completely forgot to add the improved version here. I'll add the commit tonight.

Removing `#pragma unroll` should not cause significant performance loss but one can expect 100-200MLUPS worth of difference i.e. about 1-2% change in execution times.